src/crypt.c

```diff
@@ -990,4 +1092,24 @@ crypt_sodium_buffer_decode(
 # endif
 }
 
+# if defined(FEAT_SODIUM) || defined(PROTO)
+    int
+crypt_sodium_munlock(void *const addr, const size_t len)
+{
+# if DYNAMIC_SODIUM
+    // FIXME:
+    // The free_buf_options() function in buffer.c may call this without
+    // calling crypt_sodium_init().
```

Comment:
I found that `sodium_munlock()` may be called from `free_buf_options()` without calling `crypt_sodium_init()`.

Steps to reproduce:
1. Start vim: `gvim -u DEFAULTS`
2. Add some text: `ahello<Esc>`
3. Enable libsodium: `:set cm=xchacha20`
4. Set encrypt key: `:X` (e.g. "hello")
5. Write to a file: `:w hello.txt`
6. `sodium_munlock()` is called without `crypt_sodium_init()`.

@chrisbra 
This is unintended, right?

Reply:
yes, I don't think this should happen

Reply:
The issue has been fixed: https://github.com/vim/vim/pull/9554/commits/7cb7f9c09310d451896677ebe676c9f822fa616d